packages/expo-sharing/android/src/main/java/expo/modules/sharing/SharingModule.java

```diff
@@ -0,0 +1,88 @@
+package expo.modules.sharing;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.v4.content.FileProvider;
+
+import java.io.File;
+import java.net.URLConnection;
+
+import expo.core.ExportedModule;
+import expo.core.ModuleRegistry;
+import expo.core.Promise;
+import expo.core.arguments.ReadableArguments;
+import expo.core.interfaces.ExpoMethod;
+import expo.core.interfaces.ModuleRegistryConsumer;
+
+public class SharingModule extends ExportedModule implements ModuleRegistryConsumer {
+  private static final String TAG = "ExpoSharing";
+  private static final String MIME_TYPE_OPTIONS_KEY = "mimeType";
+
+  private ModuleRegistry mModuleRegistry;
+  private Context mContext;
+
+  public SharingModule(Context context) {
+    super(context);
+    mContext = context;
+  }
+
+  @Override
+  public String getName() {
+    return TAG;
+  }
+  public ModuleRegistry getModuleRegistry() {
+    return mModuleRegistry;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+  }
+
+  @ExpoMethod
+  public void shareAsync(String url, ReadableArguments params, final Promise promise) {
```

Comment:
This is pretty much a special case of the logic in the `expo-intent-launcher` module. Have you thought about just implementing the Android side of this module as a JS wrapper around `IntentLauncher`/`FS` instead? Might need to add another FS method for getting a shareable URL from `FileProvider`, but I wonder if that approach might be better than maintaining essentially the same native logic in two different modules.

Reply:
Yes, we were thinking of using `IntentLauncher`, but my assumption was that `Sharing` module is for more specific use cases and depending on `IntentLauncher` just for a few lines of code may not make sense since it would add another dependency that the user has to install.

Reply:
Hmm, that makes sense. (It would probably also need to depend on FS => another dependency.) Just throwing this out there -- might it make sense move the more general logic from the `intent-launcher` module into here? It would be nice if we could figure out a way not to need to maintain duplicate native code. Not the end of the world if not, though.